### PR TITLE
Implemented SeriesImputer

### PR DESCRIPTION
**Pull request by gjdv**
Hi, 

I have implemented the SeriesImputer to solve [issue 43](https://github.com/kearnz/autoimpute/issues/43). It turned out to be easiest to make it a subclass of MultipleImputer. 
I needed to make minor modifications to SingleImputer, which also solve [issue 44](https://github.com/kearnz/autoimpute/issues/44). The added arguments default properly for use beyond SeriesImputer.
Here is some code you can use for testing:
```
import numpy as np
import pandas as pd
from autoimpute.imputations import MultipleImputer, SeriesImputer

n = 50
dat = pd.DataFrame({'A': np.random.uniform(0, 1, n), 'B': np.random.uniform(0, 1, n), 'C': np.random.uniform(0, 1, n)})
dat['B'][dat['B'] < 0.25] = np.nan
dat['C'][dat['C'] < 0.7] = np.nan

# imp = MultipleImputer(strategy='stochastic', return_list=True)
# dat_imp = imp.fit_transform(dat)

imp = SeriesImputer(strategy='stochastic', return_list=True)
dat_imp = imp.fit_transform(dat)
```

Kind regards,

Gert-Jan

### CODE REVIEW

File: autoimpute/imputations/dataframe/series_imputer.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+MultipleImputer returns the `n` imputed datasets. For each of these `n`
+imputations, the method (re)fits and applies imputation to the dataset `k`
+times. Typically `k` should be at least 3 to reach a stable state.
+"""
+
+from autoimpute.utils import check_nan_columns
+from .multiple_imputer import MultipleImputer
+
+
+class SeriesImputer(MultipleImputer):
```
Comment (kearnz) [inline autoimpute/imputations/dataframe/series_imputer.py]: Thoughts on naming this the `ChainedImputer` or the `MiceImputer` (or something similar)? I like `MiceImputer`, but open to ideas on naming.

Reasons I'd like to rename this class:
1. `SeriesImputer` has a different meaning in `autoimpute`. [Each imputation method](https://github.com/kearnz/autoimpute/tree/master/autoimpute/imputations/series) is implemented as a `SeriesImputer`, and `SeriesImputers` implement the `ISeriesImputer` ABC. 
2. The Imputer your proposing here makes significant improvements. Therefore, we should feature it, and I will rewrite the documentation to introduce it as the main class that people should use. A name like `MiceImputer` will make sense to new users, especially those coming from the R world.

File: autoimpute/imputations/dataframe/series_imputer.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+        # then check the value is greater than zero
+        if k_ < 1:
+            err = "k > 0. Cannot perform fewer than 1 imputation."
+            raise ValueError(err)
+
+        # otherwise set the property value for k
+        self._k = k_
+
+    def iterate_imputation(self, X, imp):
```
Comment (kearnz) [inline autoimpute/imputations/dataframe/series_imputer.py]: Can we rename this to `_iterate_imputation`? I add the prefix to indicate that this method is an internal helper method. Users of the class should not call this method directly.

**Comment by kearnz** [comment]
Hi - thanks for working on this, appreciate your effort. I went through it, and it looks very good. All the tests passed, so it looks like changes to the `SingleImputer` are good to go. Excited that we'll get this in there. 

I left a couple comments - let me know your thoughts. Once we resolve those, we'll merge the pull request.

**Comment by gjdv** [comment]
Hi, Thanks for your reaction. he name SeriesImputer I took from your issue-description, but I like MiceImputer better. Adding some reference to MICE in the class description might also be a good idea. I also agree with your other suggestions.
I noticed myself that I forgot to add to the comments/documentation the added argument imp_ixs to SingleImputer.fit and .transform.
I'm not too familiar, process-wise, with following up on such comments in a pull request. How do we go from here? Do you make these changes, or should I do that in my fork?

**Comment by kearnz** [comment]
Great - let's go with `MiceImputer`. For documentation, add any notes you think are necessary to the actual classes themselves. You can follow the format that I've used for methods in other classes. The format is important because the docs are autogenerated, but feel free to write any content within the format. Once everything is merged, I'll regenerate the documentation, and I'll make amendments to the README, bump the version, etc.

Regarding process, normally I'd suggest using `git commit --amend`, as these are pretty straightforward changes to a non-merged PR. That being said, I'm going to re-create the dev branch, so I'm going to close this PR. What you can do is make the changes we discussed here, then issue a new PR to the fresh dev branch, and we'll be good to go.

**Comment by kearnz** [comment]
New `dev` branch pushed. Did this to clean some stuff up locally, as I was dealing with some conflicts. Anyway, should be good to to issue a new pull request with changes, and we'll integrate from there. I'm sure there's a better way I could do this process wise too...

**Pull request closed without merging**